cmd/skopeo/sync.go

```diff
@@ -758,7 +770,17 @@ func (opts *syncOptions) run(args []string, stdout io.Writer) (retErr error) {
 			} else {
 				logrus.WithFields(fromToFields).Infof("Copying image ref %d/%d", counter+1, len(srcRepo.ImageRefs))
 				if err = retry.IfNecessary(ctx, func() error {
-					_, err = copy.Image(ctx, policyContext, destRef, ref, &options)
+					manifestBytes, err := copy.Image(ctx, policyContext, destRef, ref, &options)
+					if opts.digestFile != "" {
+						manifestDigest, err := manifest.Digest(manifestBytes)
+						if err != nil {
+							return err
+						}
+						_, err = digestFile.WriteString(ref.DockerReference().String() + " " + manifestDigest.String() + "\n")
```

Comment:
`ref.DockerReference()` can be `nil`, then this would crash.

I think `transports.ImageName` is the only fully general format we can use here.